sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -84,17 +84,27 @@ object BindReferences extends Logging {
       expression: A,
       input: Seq[Attribute],
```

Comment:
I think we should add an overload which takes a sequence of expressions and binds all of their references. We should then replace the call sites in the various projection operators.